xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAESink.cpp

```diff
@@ -837,6 +837,78 @@ void CActiveAESink::OpenSink()
   CLog::Log(LOGDEBUG, "CActiveAESink::OpenSink - trying to open device {}", device);
   m_sink = CAESinkFactory::Create(device, m_sinkFormat);
 
+  // try other device with same driver and same friendly name
+  // is probably the same device renamed due driver update
+  if (!m_sink && !m_sinkInfoList.empty() && !dev.driver.empty() && !dev.friendlyName.empty())
+  {
+    for (const auto& sink : m_sinkInfoList)
+    {
+      if (sink.m_sinkName != dev.driver)
+        continue;
+
+      if (m_sink)
+        break;
+
+      for (const auto& d : sink.m_deviceInfoList)
+      {
+        if (d.m_displayName == dev.friendlyName || d.m_displayNameExtra == dev.friendlyName)
+        {
+          m_deviceFriendlyName = dev.friendlyName;
+          device = dev.driver + ":" + d.m_deviceName;
+          m_sinkFormat = m_requestedFormat;
+          CLog::Log(LOGWARNING,
+                    "CActiveAESink::OpenSink - it was not possible open currently configured audio "
+                    "device '{}:{} ({})' will be tried audio device '{} ({})' instead.\n*** To fix "
+                    "this you need to open the audio settings and select an existing device ***",
+                    dev.driver, dev.name, dev.friendlyName, device, m_deviceFriendlyName);
+          CLog::Log(LOGDEBUG, "CActiveAESink::OpenSink - trying to open device {}", device);
+          m_sink = CAESinkFactory::Create(device, m_sinkFormat);
+          break;
+        }
+      }
+    }
+  }
+
+  std::string bestMatch;
+
+  // try default device of same driver or first device of same driver
+  if (!m_sink && !m_sinkInfoList.empty() && !dev.driver.empty())
+  {
+    for (const auto& sink : m_sinkInfoList)
+    {
+      if (sink.m_sinkName != dev.driver)
+        continue;
+
+      if (!bestMatch.empty())
+        break;
+
+      for (const auto& d : sink.m_deviceInfoList)
+      {
+        if (bestMatch.empty()) // save first device as fallback
+          bestMatch = d.m_deviceName;
+
+        if (d.m_deviceName.find("default") != std::string::npos)
+        {
+          bestMatch = d.m_deviceName;
+          break;
+        }
+      }
+    }
+  }
+  if (!m_sink && !bestMatch.empty())
+  {
+    GetDeviceFriendlyName(bestMatch);
```

Comment:
consistent with initial assignment to m_deviceFriendlyName but not the previous block (see comment on previous block)

Reply:
Changed all to use GetDeviceFriendlyName